src/main/java/performance/LocustReportReader.java

```diff
@@ -0,0 +1,65 @@
+package performance;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+import org.json.simple.parser.ParseException;
+
+import com.google.gson.Gson;
+
+import helpers.FileOperations;
+import model.LocustDistributionData;
+import model.LocustRequestData;
+
+
+public class LocustReportReader {
+
+    private static final String FILEPATHREQUEST= "target/csvlocustsresults/performanceResults_requests.csv";
+    private static final String FILEPATHDISTRIBUTION= "target/csvlocustsresults/performanceResults_distribution.csv";
+    private static final String ERROR = "ERROR, the file can not be found in the specified path";
+    
+    private String absoluteFilePathRequest = Paths.get(FILEPATHREQUEST).toFile().getAbsolutePath();
+	private String absoluteFilePathDistribution = Paths.get(FILEPATHDISTRIBUTION).toFile().getAbsolutePath();  
+    private List<LocustRequestData> locustRequestDataList;
+    private List<LocustDistributionData> locustDistributionDataList;
+    
+    private FileOperations fileOperations = new FileOperations();
+    private Gson gson = new Gson();
+
+    public LocustReportReader() {
+        this.locustRequestDataList = getLocustRequestData();
+        this.locustDistributionDataList = getLocustDistributionData();
+    }
+    
+
+    public List<LocustRequestData> getLocustRequestDataList() {
+        return this.locustRequestDataList;
+    }
+
+    public List<LocustDistributionData> getLocustDistributionDataList() {
+        return this.locustDistributionDataList;
+    }
+
+    private List<LocustRequestData> getLocustRequestData() {
+        try {
+            LocustRequestData[] locustData = gson.fromJson(fileOperations.CSVToJson(absoluteFilePathRequest).toJSONString(), LocustRequestData[].class);
+            return Arrays.asList(locustData);
+        } catch (IOException | ParseException e) {
+            throw new RuntimeException(ERROR);
+        }
+    }
+
+    private List<LocustDistributionData> getLocustDistributionData() {
```

Comment:
Why do we need to convert this to Json?

Reply:
This is answered in another questions ... (we pass the information to json in order to create an object with gson)

Reply:
But why do we need it?

Reply:
And why not? (this is a commodity to manage the data) Anyway, we can read directly from the csv and store the data in an ArrayList, then us this array to recover the data. This could be another way ... (but we will need to iterate or search for the data ...). Do you find this way better?